app/scripts/streams.js

```diff
@@ -0,0 +1,121 @@
+/**
+ *  RxJS in action
+ *  Chapter #
+ *  @author Paul Daniels
+ *  @author Luis Atencio
+ */
+
+const {createStore} = Redux;
+const {ofType, createDispatcher} = DispatcherUtils;
+
+const dispatcher = createDispatcher(createStore(bankingApp, {
+  accounts: {checking: 100, savings: 100},
+  transactions: {transactions: [], limit: 10, skip: 0},
+  messages: [],
+  searches: {query: '', results: []}
+}));
+
+const balanceActions = {
+  withdraw: () => dispatcher.dispatch({type: PROCESS_TRANSACTION, factor: -1}),
+  deposit: () => dispatcher.dispatch({type: PROCESS_TRANSACTION, factor: 1}),
+  amount: (value) => dispatcher.dispatch({type: UPDATE_AMOUNT, amount: value}),
+  account: (value = 'checking') => dispatcher.dispatch({type: UPDATE_ACCOUNT, account: value})
+};
+
+dispatcher.include(handleSearch);
+dispatcher.include(handleMessages);
+dispatcher.include(processUserTransaction);
+dispatcher.include(processInterest);
+dispatcher.include(handleTransaction);
+
+function handleMessages(actions$, store$) {
+  return Rx.Observable.interval(250)
+    .withLatestFrom(store$.pluck('messages'))
+    .filter(([_, messages]) => messages.length > 0)
+    .mapTo(degradeMessages());
+}
+
+function handleSearch(action$, state$) {
+  return action$
+    .let(ofType(INVOKE_SEARCH))
+    .debounceTime(500)
+    .switchMap(
+      ({query}) => search(query, {limit: 20})
+    ).map(
+      results => updateSearchResults(results)
+    )
+}
+
+function processUserTransaction(actions$) {
+  const do$ = actions$.let(ofType(PROCESS_TRANSACTION)).pluck('factor');
+  const amount$ = actions$.let(ofType(UPDATE_AMOUNT))
+    .pluck('amount')
+    .map(x => typeof x === 'number' ? x : parseInt(x));
+  const account$ = actions$.let(ofType(UPDATE_ACCOUNT)).pluck('account');
+
+  return do$.withLatestFrom(amount$, account$, (factor, amount, account) =>
+    ({amount, account, factor, type: NEW_TRANSACTION}))
+}
```

Comment:
Here, splitting out several streams from the `action$` stream and then using them to create a new transaction